### PR TITLE
removing favorites & bid list from similar positions results

### PR DESCRIPTION
**Pull request by SidelineCory24**
Initial testing/help code. [BE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/412)

### CODE REVIEW

**Comment by SidelineCory24** [comment]
pardon the comments/test code

**Comment by elizabeth-jimenez** [comment]
> pardon the comments/test code

No, no; this is exactly what a WIP should look like!

File: src/actions/results.js [hunk 1]
```diff
@@ -63,11 +65,50 @@ export function resultsFetchSimilarPositions(id) {
...
+        }
+        console.log(favoritesBidListArray);
+        console.log(results);
+        // comparison check: filter/lodash
+        // filter(users, o => { !o.active;});
+        // const filteredUsers = filter(users, a => !a.active);
+        // const filteredUsers = filter(results.results, a => console.log(a.id));
+        // try teneray operator for comparison if possible
+        const filteredPositions = filter(filteredResults, (a) => {
+          if (!favoritesBidListArray.includes(a.id)) { return true; } return false;
```
Comment (elizabeth-jimenez) [inline src/actions/results.js:87]: Ahh! I knew we didn't need an 'if', i just got caught up in the syntax 😅 

changing line 87 to just `!favoritesBidListArray.includes(a.id)` should work. May or may not need to put a `return` in front of it.

File: src/actions/results.js [hunk 2]
```diff
@@ -63,11 +65,50 @@ export function resultsFetchSimilarPositions(id) {
...
+        // payload master array favs/bidList of ids
+        const favoritesBidListArray = [];
+        for (let f = 0; f < favorites.length; f += 1) {
+          favoritesBidListArray.push(Number(favorites[f].id));
+          // console.log(favorites[f].id);
+        }
+        for (let b = 0; b < bidList.length; b += 1) {
+          favoritesBidListArray.push(bidList[b].position.id);
+          // console.log(bidList[b].position.id);
+        }
```
Comment (elizabeth-jimenez) [inline src/actions/results.js:78]: Try to see if you can convert these `for` loops to use `forEach`.

The section **Converting a for loop to forEach** at [mozilla](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) is a good reference.

File: src/actions/results.js (hunk 1 above)
Comment (SidelineCory24) [reply src/actions/results.js]: done thanks🙏

File: src/actions/results.js (hunk 2 above)
Comment (SidelineCory24) [reply src/actions/results.js]: and another one😃

**Comment by elizabeth-jimenez** [comment]
@SidelineCory24 @scott062 I tested this more and it seems to be just fine, actually. It could be that our latest changes just weren't taking effect

🤷‍♀️ 

So only thing left is to confirm that it is to clean it up, confirm it is working, and fix the test. Make sure to also test adding to bidlist to make sure those also filter out and trigger original results if was the last item.

**Comment by SidelineCory24** [comment]
final review/testing/code cleanup is complete

**Review by elizabeth-jimenez — APPROVED** [review]
Good job Cory! 🍬 🍰 🍭 🥮 👍

**Pull request merged**